### PR TITLE
Feature/display

### PR DESCRIPTION
**Pull request by bronexproduction**
stdgunw:
* Disabled `cxxabi` compilation, minor Makefile updates
* NULL, nullptr, true, false definitions
* mem.c compilation tweaks

System:
* Display driver
* Terminal now accesses video memory accessible via device driver (display)
* New syscalls: 
  * DEV_GET_BY_ID
  * DEV_GET_BY_TYPE
  * DEV_ACQUIRE
  * DEV_RELEASE
  * DEV_WRITE
  * FUG
* GunwDisplayAPI
* New UHA types for display and memory-mapped devices
* UHA split into descriptor (public) and routines (private)
* Removed some circular dependencies and upper-to-lower ring dependencies
* More restrictive compilation flags
* Minor updates and fixes

### CODE REVIEW

File: src/lib/stdgunw/mem.c [hunk 1]
```diff
@@ -8,13 +8,15 @@
...
+    uint_8 *p = (uint_8 *)ptr;
+    while (bytes--) {
+        *(p++) = val;
     }
 }
 
 void memnull(void *ptr, size_t bytes) {
-    return memset(ptr, 0, bytes);
+    memset(ptr, 0, bytes);
+    return;
```
Comment (merskip) [inline src/lib/stdgunw/mem.c:19]:
```suggestion
```

File: src/system/api/include/gunwdev.h [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+};
+
+/*
+    Requests installation of given hardware driver
+
+    Params:
+        * id - device identifier 
+               to be returned if the installation succeeds
+        * desc - device descriptor (see gunwdrv.h)
+*/
```
Comment (merskip) [inline src/system/api/include/gunwdev.h:34]: Czy mógłbyś jeszcze dorzucić opis wartości zwracanej?

File: src/system/api/src/_gunwdisplay.h [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+#ifndef GUNWOS__GUNWDISPLAY_H
+#define GUNWOS__GUNWDISPLAY_H
+
+#include <gunwdisplay.h>
+#include <gunwdev.h>
+#include <gunwfug.h>
+#include <stdgunw/mem.h>
+#include <scl_def.h>
+
+enum displayType {
```
Comment (merskip) [inline src/system/api/src/_gunwdisplay.h:17]:
```suggestion
enum DisplayType {
```

**Review by merskip — APPROVED** [review]
w pytke, dobra robota

File: src/system/api/src/_gunwdisplay.h (hunk 3 above)
Comment (bronexproduction) [reply src/system/api/src/_gunwdisplay.h:17]: Applied

File: src/lib/stdgunw/mem.c (hunk 1 above)
Comment (bronexproduction) [reply src/lib/stdgunw/mem.c]: Applied

**Pull request merged**